build.rs

```diff
@@ -0,0 +1,116 @@
+use crate::SqlType::{Idempotent, Migration};
+use askama::Template;
+use std::fs::File;
+use std::io::{Error, Read, Write};
+use std::path::{Path, PathBuf};
+use std::string::String;
+use std::{env, fs};
+
+#[derive(Template)]
+#[template(path = "migration-wrapper.sql", escape = "none")]
+struct MigrationWrapper<'a> {
+    filename: &'a str,
+    version: &'a str,
+    body: String,
+}
+
+#[derive(Template)]
+#[template(path = "idempotent-wrapper.sql", escape = "none")]
+struct IdempotentWrapper<'a> {
+    filename: &'a str,
+    body: String,
+}
+
+fn main() {
+    println!("cargo:rerun-if-changed=build.rs");
+    println!("cargo:rerun-if-changed=Cargo.lock");
+    println!("cargo:rerun-if-changed=migration");
+    println!("cargo:rerun-if-changed=templates");
+    println!("cargo:rerun-if-changed=migration-table.sql");
+
+    let mut sql = include_str!("migration-table.sql").to_string();
+    let rendered_sql = render_sql().expect("unable to render migrations");
+    sql.push_str(rendered_sql.as_str());
+
+    let manifest_dir = env::var("CARGO_MANIFEST_DIR").unwrap();
+    let mut out_path = PathBuf::from(manifest_dir);
+    out_path.push("migration.sql");
```

Comment:
nit: migration.sql seems too generic. Maybe generated-migration.sql or hand-written-migration.sql?

Reply:
I like hand-written-migration. It's long, but it's explicit.